From ddb48f9c68684c426bdd362b9323310224ba9ea4 Mon Sep 17 00:00:00 2001
From: Wayne Zou <b36644@freescale.com>
Date: Thu, 21 Jun 2012 12:45:52 +0800
Subject: [PATCH 325/505] ENGR00209057 ipudev unit test: Use one dma FB buffer
 when show image on fb

Reduce 3 FB buffers to one FB buffer when show image on fb,
so it will not consume a lot of system memory after long test.
---
 test/mxc_ipudev_test/mxc_ipudev_test.c | 15 +++++----------
 1 file changed, 5 insertions(+), 10 deletions(-)

diff --git a/test/mxc_ipudev_test/mxc_ipudev_test.c b/test/mxc_ipudev_test/mxc_ipudev_test.c
index 83c2776..c290da2 100644
--- a/test/mxc_ipudev_test/mxc_ipudev_test.c
+++ b/test/mxc_ipudev_test/mxc_ipudev_test.c
@@ -34,7 +34,6 @@
 #include <linux/mxcfb.h>
 #include "mxc_ipudev_test.h"
 
-#define FB_BUFS 3
 #define PAGE_ALIGN(x) (((x) + 4095) & ~4095)
 int ctrl_c_rev = 0;
 void ctrl_c_handler(int signum, siginfo_t *info, void *myact)
@@ -174,7 +173,7 @@ int main(int argc, char *argv[])
 	void *inbuf = NULL, *vdibuf = NULL;
 	void *ovbuf = NULL, *alpbuf = NULL;
 	void *outbuf = NULL;
-	dma_addr_t outpaddr[FB_BUFS];
+	dma_addr_t outpaddr;
 	struct fb_var_screeninfo fb_var;
 	struct fb_fix_screeninfo fb_fix;
 	int blank;
@@ -336,8 +335,9 @@ int main(int argc, char *argv[])
 		fb_var.xres = t->output.width;
 		fb_var.xres_virtual = fb_var.xres;
 		fb_var.yres = t->output.height;
-		fb_var.yres_virtual = fb_var.yres * FB_BUFS;
+		fb_var.yres_virtual = fb_var.yres;
 		fb_var.activate |= FB_ACTIVATE_FORCE;
+		fb_var.vmode |= FB_VMODE_YWRAP;
 		fb_var.nonstd = t->output.format;
 		fb_var.bits_per_pixel = fmt_to_bpp(t->output.format);
 
@@ -349,10 +349,7 @@ int main(int argc, char *argv[])
 		ioctl(fd_fb, FBIOGET_VSCREENINFO, &fb_var);
 		ioctl(fd_fb, FBIOGET_FSCREENINFO, &fb_fix);
 
-		for (i=0; i<FB_BUFS; i++)
-			outpaddr[i] = fb_fix.smem_start +
-				i * fb_var.yres * fb_fix.line_length;
-
+		outpaddr = fb_fix.smem_start;
 		blank = FB_BLANK_UNBLANK;
 		ioctl(fd_fb, FBIOBLANK, blank);
 	} else {
@@ -432,7 +429,7 @@ task_begin:
 		}
 
 		if (test_handle.show_to_fb)
-			t->output.paddr = outpaddr[done_cnt % FB_BUFS];
+			t->output.paddr = outpaddr;
 
 		ret = ioctl(fd_ipu, IPU_QUEUE_TASK, t);
 		if (ret < 0) {
@@ -441,8 +438,6 @@ task_begin:
 		}
 
 		if (test_handle.show_to_fb) {
-			fb_var.vmode &= ~FB_VMODE_YWRAP;
-			fb_var.yoffset = (done_cnt % FB_BUFS) * fb_var.yres;
 			ret = ioctl(fd_fb, FBIOPAN_DISPLAY, &fb_var);
 			if (ret < 0) {
 				printf("fb ioct FBIOPAN_DISPLAY fail\n");
-- 
1.8.0

